plugins/catalog-import/src/components/ImportStepper/defaults.tsx

```diff
@@ -71,16 +71,29 @@ export type StepperProvider = {
   ) => StepConfiguration;
 };
 
+/**
+ * The types that are creatable from a repository which does not yet include a catalog-info.yaml file
+ *
+ * @see https://backstage.io/docs/features/software-catalog/descriptor-format#spectype-required
+ */
+enum CreatableFromRepositoryEntityType {
+  service = 'service',
+  website = 'website',
+  library = 'library',
+  documentation = 'documentation',
+  other = 'other',
+}
```

Comment:
I think this should be configurable, especially as the Backstage documentation itself thinks that using `other` [is not a good idea](https://backstage.io/docs/features/software-catalog/extending-the-model#adding-a-new-type-of-an-existing-kind).

Reply:
100% agree. Having these hard-coded is going to be a non starter for lots of people. @Rugvip mentioned something about maybe added a proper API around exposing the allowed/recommended list for an org maybe in the `app-config`. This could most likely use that approach.

Reply:
Also, I think another comment touches on this, but there will be different type sets for different entity kinds (e.g. `API => (openapi, graphql, asyncapi, etc.)`)

Reply:
Something like this? We could have an "allow" approach which whitelists types or a "recommended" approach to suggest types in this form. Or both but maybe a bit complex.

If the configuration includes `allow` for the `Component` kind, the user will only be able to select one of these types and won't be able to freely type. If the configuration includes `recommended` for the `Component` kind, the user will have those recommended but will be able to freely type something else. If both are present, the select will suggest the intersection of both and the user will not be able to write something else.

<img width="250" alt="Capture d’écran 2021-02-27 à 12 19 11" src="https://user-images.githubusercontent.com/4517991/109385690-05d40100-78f6-11eb-9284-72c3f20e3767.png">

Reply:
Allow for recommended might be different for each org. I'd say starting with a recommend list makes most sense. I've come up with a few cases where I'm entirely sure how to classify something under the existing recommended types and end up throwing them in `other` for now (with the intention of cleaning these up later. Some examples are repos that contain IaaC, docker images, demos, cloud functions, etc.

Perhaps something like this could provide a baseline + room for extension?

```yaml
catalog:
  kinds:
    Component: # not sure on upper-casing this here, but having this keyed would be nice
       typeRule: 'recommended' # or 'restricted'
       types:
         - service
         - website
         - library
         - documentation
```

Reply:
Having a bit of trouble thinking about how we make this a more general thing, avoiding things like `Component` entities having a `type`. The `type` isn't something that applies to all entities, and some entities might have their own version, either called `type` or something else. Was thinking that something like JSONSchema is already able to express all we need, but that also gets really verbose.

I'm thinking we can head down two paths, either we (1) standardize on `spec.type` being a known field that entities can use to further distinguish between things, or we (2) create a dead simple and limited schema for expressing possible values of an entity field, something like

```yaml
kinds:
  Component:
    fields:
    - path: 'spec.type'
      values:
      - service
      - website
      - library
      - documentation
```

Personally I'm leaning towards (1) as I think it could provide benefit in a bunch of places, but I'm also immediately worried that the need for a 3rd dimension pops up and we're soon dealing with hyperentities. There's also the case that settling (2) would let us describe things that are perhaps not quite as important to how the frontend renders things, but it still nice to put some bounds on, like `spec.lifecycle`.